tmail-backend/integration-tests/jmap/distributed-jmap-integration-tests/src/test/java/com/linagora/tmail/james/DistributedLinagoraSecondaryBlobStoreTest.java

```diff
@@ -0,0 +1,192 @@
+import com.linagora.tmail.blob.guice.BlobStoreConfiguration;
+import com.linagora.tmail.encrypted.MailboxConfiguration;
+import com.linagora.tmail.james.app.CassandraExtension;
+import com.linagora.tmail.james.app.DistributedJamesConfiguration;
+import com.linagora.tmail.james.app.DistributedServer;
+import com.linagora.tmail.james.app.EventBusKeysChoice;
+import com.linagora.tmail.james.app.RabbitMQExtension;
+import com.linagora.tmail.james.common.EncryptHelper;
+import com.linagora.tmail.james.common.LinagoraEmailSendMethodContract$;
+import com.linagora.tmail.james.common.module.JmapGuiceKeystoreManagerModule;
+import com.linagora.tmail.module.LinagoraTestJMAPServerModule;
+
+import io.restassured.authentication.PreemptiveBasicAuthScheme;
+import io.restassured.builder.RequestSpecBuilder;
+import io.restassured.config.EncoderConfig;
+import io.restassured.config.RestAssuredConfig;
+import reactor.core.publisher.Flux;
+import reactor.util.retry.Retry;
+
+public class DistributedLinagoraSecondaryBlobStoreTest {
+    static class BlobStoreProbe implements GuiceProbe {
+        private final BlobStoreDAO firstBlobStoreDAO;
+        private final BlobStoreDAO secondBlobStoreDAO;
+
+        @Inject
+        public BlobStoreProbe(@Named(FIRST_LEVEL) BlobStoreDAO firstBlobStoreDAO,
+                              @Named(SECONDARY_BLOB_STORE_DAO) BlobStoreDAO secondBlobStoreDAO) {
+            this.firstBlobStoreDAO = firstBlobStoreDAO;
+            this.secondBlobStoreDAO = secondBlobStoreDAO;
+        }
+
+        public BlobStoreDAO getFirstBlobStoreDAO() {
+            return firstBlobStoreDAO;
+        }
+
+        public BlobStoreDAO getSecondBlobStoreDAO() {
+            return secondBlobStoreDAO;
+        }
+    }
+
+    static final Domain DOMAIN = Domain.of("domain.tld");
+    static final Username BOB = Username.fromLocalPartWithDomain("bob", DOMAIN);
+    static final Username ANDRE = Username.fromLocalPartWithDomain("andre", DOMAIN);
+    static final String BOB_PASSWORD = "bobpassword";
+    static final String ANDRE_PASSWORD = "andrepassword";
+    static final String ACCEPT_RFC8621_VERSION_HEADER = "application/json; jmapVersion=rfc-8621";
+    static final String ACCOUNT_ID = "29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6";
+
+    static DockerAwsS3Container dockerAwsS3Container = new DockerAwsS3Container();
+    static S3BlobStoreConfiguration secondaryS3Configuration;
+
+    static {
+        dockerAwsS3Container.start();
+        AwsS3AuthConfiguration authConfiguration = AwsS3AuthConfiguration.builder()
+            .endpoint(dockerAwsS3Container.getEndpoint())
+            .accessKeyId(DockerAwsS3Container.ACCESS_KEY_ID)
+            .secretKey(DockerAwsS3Container.SECRET_ACCESS_KEY)
+            .build();
+
+        secondaryS3Configuration = S3BlobStoreConfiguration.builder()
+            .authConfiguration(authConfiguration)
+            .region(dockerAwsS3Container.dockerAwsS3().region())
```

Comment:
Just to make sure, that's a different region from the on configured in default blob.properties file correct?

Reply:
does it matter? there are two different s3 containers in this test